### DISCUSSION TITLE
about heat conduction transfer boundary

### DISCUSSION THREAD

**Discussion by amosaha** [idaholab/moose#20162] [Q&A General] [answered]
Hi, I made a heat transfer model. The result is shown in the figure. I used the boundary condition of heatcondutionBC between the two layers of materials. I wonder if it is reasonable. Since my heat transfer result is not ideal, I wonder if I need to add other conditions? For example, there seems to be no heat transfer in the internal pellets. Also, what is the difference between ADHeat Conduction and HeatConduction？
![C87D6176-AC15-43A7-8AA6-CCC9BAB0E74C](https://user-images.githubusercontent.com/93416273/151573308-88de95b0-bf4b-4ffd-8f47-9692465ffc0e.jpeg)

**Comment by GiudGiud** [comment, marked as answer]
Hello

It looks weird? The temperature seems to be oscillating. 
Do you need a BC in between the two layers? If you wanted to model a special interface / a thin heat resistance from the contact between the two layers, I would use an interface kernel.
If you just want to model regular thermal conduction, then there is a no need for a boundary condition, the kernels already apply the right equation at the boundary between the two materials.

The difference between ADHeatConduction and HeatConduction is automatic differentiation (AD) in the first one. AD is used to compute numerically perfect derivatives for the Jacobian, especially with complex material properties (heat conductivity here). 
See more information here: 
https://mooseframework.inl.gov/moose/modules/phase_field/FunctionMaterials/AutomaticDifferentiation.html

Guillaume

**Comment by amosaha** [comment]
Thank you. I have another question. Why is there a negative value in heat transfer? That doesn't seem to fit the laws of physics.






发自我的iPhone


------------------ Original ------------------
From: Guillaume Giudicelli ***@***.***&gt;
Date: Fri,Jan 28,2022 11:41 PM
To: idaholab/moose ***@***.***&gt;
Cc: amosaha ***@***.***&gt;, Author ***@***.***&gt;
Subject: Re: [idaholab/moose] about heat conduction transfer boundary(Discussion #20162)





 
Hello
 
It looks weird? The temperature seems to be oscillating.
 Do you need a BC in between the two layers? If you wanted to model a special interface / a thin heat resistance from the contact between the two layers, I would use an interface kernel.
 If you just want to model regular thermal conduction, then there is a no need for a boundary condition, the kernels already apply the right equation.
 
The difference between ADHeatConduction and HeatConduction is automatic differentiation (AD) in the first one. AD is used to compute numerically perfect derivatives for the Jacobian, especially with complex material properties (heat conductivity here).
 See more information here:
 https://mooseframework.inl.gov/moose/modules/phase_field/FunctionMaterials/AutomaticDifferentiation.html
 
Guillaume
 
—
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.

**Reply by GiudGiud** [reply]
The problem is probably not correctly defined. The solution solves the equations that it is being fed, if there's issues with them the solution can be unphysical.
